utils/src/errors.ts

```diff
@@ -7,13 +7,18 @@ import { ITreeItemPickerContext } from "..";
 import { localize } from "./localize";
 
 export class UserCancelledError extends Error {
+    _isUserCancelledError = true;
     public stepName: string | undefined;
     constructor(stepName?: string) {
         super(localize('userCancelledError', 'Operation cancelled.'));
         this.stepName = stepName;
     }
 }
 
+export function isUserCancelledError(error: unknown): error is UserCancelledError {
+    return (error as UserCancelledError)._isUserCancelledError;
```

Comment:
Just to be slightly more resilient:

```suggestion
    return typeof error === 'object' && (error as UserCancelledError)._isUserCancelledError;
```